### DISCUSSION TITLE
Exception Stack Trace field showing namespace in the log entry instead of stack trace

### DISCUSSION THREAD

**Discussion by adityavgpl** [jongpie/NebulaLogger#366] [Q&A]
Hi All,

In my org I am using Nebula's managed package version. I am using it to log errors in my own managed package with namespace 'DeltaLKU'. But here the problem is I am not getting the exception stack trace in the log entry, instead of stack trace its showing namespace in brackets like (DeltaLKU)
![image](https://user-images.githubusercontent.com/64599355/190312411-b3068d10-29d8-46e1-80eb-09fe52a195fc.png)

Can any one help me resolve this

@jongpie do you have any suggestion for resolving this ? do we have some method to set just the stack trace string in the log entry, something like for example `setStackTrace` similar to `setExceptionDetails`

**Comment by jongpie** [comment]
Hi @adityavgpl - that might be a platform limitation with managed packages, some details about managed package code is often obfuscated like this. You could try to get around the issue by setting the exception stack trace field yourself, something like this:

```java
Exception myException; // assuming you have an exception variable called myException
Logger.error('Something broke', myException)?.getLogEntryEvent().ExceptionStackTrace__c = myException.getStackTraceString();
```

I'm not 100% sure that this approach will work, but it's definitely worth trying. Let me know if this ends up showing the correct stack trace.

**Reply by adityavgpl** [reply]
@jongpie I tried this but getting an error:
`Method is not visible: Nebula__LogEntryEvent__e Nebula.LogEntryEventBuilder.getLogEntryEvent()`


Looks like `getLogEntryEvent` is a public method, its not accessible from the apex class inside my package.

I am trying to set the stack trace field, but not getting a proper way to do it. Do you have any other suggestions that I can try?

**Reply by jongpie** [reply]
@adityavgpl my mistake! I forgot that `getLogEntryEvent()` is only public/not accessible for the managed package. That is definitely going to limit the options without releasing a new version of the managed package, so I may still ultimately need to make some changes to better support this.

Could you try one more thing though? I want to confirm that there is a way to get an accurate exception stack trace from your managed package - could you try logging just the exception stack trace string? Something like this:

```java
Exception myException; // assuming you have an exception variable called myException
Nebula.Logger.error('Something broke', myException);
Nebula.Logger.error('The exception stack trace string: ' + myException.getStackTraceString());
Nebula.Logger.saveLog();
```

This will create an extra `Nebula__LogEntry__c` record, and hopefully the `Nebula__Message__c` field will have the full stack trace of your exception.

**Reply by adityavgpl** [reply]
Tried this @jongpie . But myException.getStackTraceString() is also masking the string and just providing DeltaLKU.

Did anyone report similar issue, I know this is not an issue with Nebula Logger but just curious how proper logs are created when the error/exception is from another managed package?

**Reply by jamessimone** [reply]
I don’t think this is possible without a managed package specifically exposing a global method to get the current inner stacktrace string, since Salesforce intentionally masks the managed package stacktrace with the package name.

**Reply by jongpie** [reply]
@adityavgpl I think that means that there's not a way to get a managed package's stack trace within Nebula Logger, since (as @jamessimone mentioned), Salesforce is intentionally masking it. I have not personally implemented Nebula Logger within another managed package, so I don't know if there are any other workarounds, but since the stack trace is masked even when you call `myException.getStackTraceString()`, I don't know of any other way to try to get the stack trace. I think this is something that unfortunately won't work for managed packages.
